src/version.rs

```diff
@@ -194,6 +249,72 @@ impl Version {
                           &[&yanked, &self.id]));
         Ok(())
     }
+
+    pub fn store_build_info(&self,
+                            conn: &GenericConnection,
+                            info: upload::VersionBuildInfo,
+                            krate: &Crate) -> CargoResult<()> {
+
+        // Verify specified Rust version will parse before doing any inserting
+        let channel_version = info.channel_version()?;
+
+        // Future improvement: allow overwriting an existing row, in which case
+        // the ON CONFLICT DO should set `passed` instead of doing nothing.
+        let inserted = conn.execute("INSERT INTO build_info \
+                          (version_id, rust_version, target, passed) \
+                      VALUES ($1, $2, $3, $4) \
+                      ON CONFLICT (version_id, rust_version, target) DO NOTHING",
+            &[&self.id, &info.rust_version, &info.target, &info.passed])?;
+
+        if inserted == 0 {
+            return Err(human(format!(
+                "Build info already specified for {} v{} with {} and target {}",
```

Comment:
Ah, we can change this to make overwriting the default, definitely.

I think we were thinking about the case when you're working on the next version of a crate on the master branch, but you haven't bumped the version yet, you wouldn't want to overwrite the released version's build results with the master branch. We hadn't yet gotten to the realization that you could run things on travis on the push of a new tag... as long as everyone sets their CI up that way, overwriting should be fine. 

And then that gets rid of the need to add an `--overwrite` flag to cargo, so I'm into it!